pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -162,9 +163,14 @@ func (h *upgradeHandler) OnChanged(_ string, upgrade *harvesterv1.Upgrade) (*har
 			toUpdate.Status.UpgradeLog = upgradeLog.Name
 		}
 		harvesterv1.LogReady.CreateUnknownIfNotExists(toUpdate)
+		harvesterv1.LogReady.LastUpdated(toUpdate, time.Now().UTC().Format(time.RFC3339))
 		return h.upgradeClient.Update(toUpdate)
 	}
 
+	if harvesterv1.LogReady.IsUnknown(upgrade) {
+		return h.checkLogReadyCondition(upgrade)
+	}
+
```

Comment:
Did you check: when logReady is timeouted, what's the upgrade (UI) status?

Will the upgrade be stopped or continued? if continuing, is the UI menu `download upgrade log` still available?

L174 needs to be refactored to add a check of `upgrade.Spec.LogEnabled` first, and handle the success, failure case